File: leetcode-algorithms/024. Swap Nodes in Pairs/solution.cpp
struct ListNode {
	int val;
	ListNode *next;
	ListNode(int x) : val(x), next(NULL) {}
};


ListNode* swapPairs(ListNode* head) {
	ListNode dummy(0);
	dummy.next = head;
	ListNode *curr = &dummy;

	while(curr->next && curr->next->next)
	{
		ListNode* first = curr->next;
		ListNode* second = curr->next->next;
		first->next = second->next;
		curr->next = second;
		curr->next->next = first;
		curr = curr->next->next;
	}
	return dummy.next;
}

File: leetcode-algorithms/024. Swap Nodes in Pairs/solution.py
class ListNode:
	def __init__(self, x):
		self.val = x
		self.next = None

def swapPairs(head):
	"""
	:type head: ListNode
	:rtype: ListNode
	"""

	dummy = ListNode(None)
	dummy.next = head
	curr = dummy

	while curr.next and curr.next.next:
		first = curr.next
		second = curr.next.next
		first.next = second.next
		curr.next = second
		curr.next.next = first
		curr = curr.next.next
	return dummy.next